helix-term/src/ui/editor.rs

```diff
@@ -248,6 +257,103 @@ impl EditorView {
             .for_each(|area| surface.set_style(area, ruler_theme))
     }
 
+    /// Gets the word under the cursor
+    pub fn cursor_word(doc: &Document, view: &View) -> Option<String> {
+        let text = doc.text().slice(..);
+        let cursor = doc.selection(view.id).primary().cursor(text);
+        let char_under_cursor = text.get_char(cursor);
+        if !char_under_cursor.map_or(false, char_is_word) {
+            return None;
+        }
+
+        let chars_at_cursor = text.chars_at(cursor);
+        let reversed_chars = chars_at_cursor.clone().reversed();
+        let start = cursor.saturating_sub(reversed_chars.take_while(|c| char_is_word(*c)).count());
+        let end = cursor + chars_at_cursor.take_while(|c| char_is_word(*c)).count();
+
+        Some(text.slice(start..end).to_string())
```

Comment:
Is returning a `String` necessary here ? I think it could be a `&str` reference to the document and only converted to a `String` when the lifetime become annoying

Reply:
returning a `String` is not necessary. Thanks for finding that one! :)